openml/flows/flow.py

```diff
@@ -326,16 +352,27 @@ def publish(self):
         # instantiate an OpenMLFlow.
         import openml.flows.functions
 
-        xml_description = self._to_xml()
+        flow_id = openml.flows.functions.flow_exists(self.name, self.external_version)
+        if not flow_id:
```

Comment:
I think this would be simpler to understand if you control with self.flow_id and not flow_id. It organizes the checks a bit nicer in my opinion.

Reply:
I am not sure I follow. `flow_id` and `self.flow_id` are not the same. In this case `flow_id` is `False` if the flow does not yet exist on the server, or an integer representing the flow id as known by the server. `self.flow_id` is None if it has not yet been synced with the server, or an integer if it has. Or do you mean changing it to:
```
if self.flow_id and not flow_id:
    raise ...
elif not flow_id:
    ...
    rest of code
```
?

Reply:
I meant the second case that you gave a code example for and I was thinking of a different organization. But looking at it more, I do not see a clear way on how to not have extra checks.